include/deal.II/fe/mapping_manifold.h

```diff
@@ -0,0 +1,581 @@
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_second_derivatives.size()));
+//   return shape_second_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+
+// template <int dim, int spacedim>
+// inline
+// Tensor<2,dim> &
+// MappingManifold<dim,spacedim>::InternalData::second_derivative (const unsigned int qpoint,
+//     const unsigned int shape_nr)
+// {
+//   Assert(qpoint*n_shape_functions + shape_nr < shape_second_derivatives.size(),
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_second_derivatives.size()));
+//   return shape_second_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+// template <int dim, int spacedim>
+// inline
+// const Tensor<3,dim> &
+// MappingManifold<dim,spacedim>::InternalData::third_derivative (const unsigned int qpoint,
+//     const unsigned int shape_nr) const
+// {
+//   Assert(qpoint*n_shape_functions + shape_nr < shape_third_derivatives.size(),
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_third_derivatives.size()));
+//   return shape_third_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+
+// template <int dim, int spacedim>
+// inline
+// Tensor<3,dim> &
+// MappingManifold<dim,spacedim>::InternalData::third_derivative (const unsigned int qpoint,
+//     const unsigned int shape_nr)
+// {
+//   Assert(qpoint*n_shape_functions + shape_nr < shape_third_derivatives.size(),
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_third_derivatives.size()));
+//   return shape_third_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+
+// template <int dim, int spacedim>
+// inline
+// const Tensor<4,dim> &
+// MappingManifold<dim,spacedim>::InternalData::fourth_derivative (const unsigned int qpoint,
+//     const unsigned int shape_nr) const
+// {
+//   Assert(qpoint*n_shape_functions + shape_nr < shape_fourth_derivatives.size(),
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_fourth_derivatives.size()));
+//   return shape_fourth_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+
+// template <int dim, int spacedim>
+// inline
+// Tensor<4,dim> &
+// MappingManifold<dim,spacedim>::InternalData::fourth_derivative (const unsigned int qpoint,
+//     const unsigned int shape_nr)
+// {
+//   Assert(qpoint*n_shape_functions + shape_nr < shape_fourth_derivatives.size(),
+//          ExcIndexRange(qpoint*n_shape_functions + shape_nr, 0,
+//                        shape_fourth_derivatives.size()));
+//   return shape_fourth_derivatives [qpoint*n_shape_functions + shape_nr];
+// }
+
+
+template <int dim, int spacedim>
+inline
+void
+MappingManifold<dim,spacedim>::InternalData::compute_manifold_quadrature_weights (const Quadrature<dim> &quad)
+{
+  FE_Q<dim> fe_q(1);
```

Comment:
Also, this class already has a member variable of this type -- definitely no need to create another one here :-)